Refactor/speaker cards session content

ShugKnight24 opened the pull request:
# Pull Request

## Description

- ~~Update SpeakerCards to match new ProfileCard design~~
- Update a few speaker images to work with new design
- Add placeholder when a track has no current sessions
- Add session descriptions
- Add speaker description
- Add partner description
- Add WTM logo for panelists
- Add icon for GDEs
- Update ProfileCard display for speakers
- Add speaker urls to the SpeakerDetails

## Type of Change

- [X] New feature (non-breaking change which adds functionality)
- [X] Code refactoring

## Related Issues

- Closes #77 
- Closes #80 

## Code Quality Checklist

- [X] Code follows the project's coding standards
- [X] Self-review of the code has been performed
- [X] Code is properly commented where necessary
- [X] No console.log statements left in the code
- [X] No hardcoded values that should be configurable
- [X] Error handling is implemented where appropriate

## Testing

- [X] Changes have been tested locally
- [X] All existing tests pass (if applicable)
- [X] New tests have been added for new functionality (if applicable)
- [X] Manual testing has been performed
- [X] Accessibility testing has been performed (for UI changes)

## Accessibility (for UI changes)

- [X] All interactive elements are keyboard accessible
- [X] Color contrast meets WCAG AA standards
- [X] Images have appropriate alt text
- [X] Form labels are properly associated
- [X] Focus indicators are visible
- [X] Screen reader compatibility has been tested

## Build & Deployment

- [X] Application builds successfully
- [X] No build warnings or errors
- [X] Dependencies are up to date
- [X] No security vulnerabilities introduced

## Screenshots (if applicable)

### Session description text

<img width="1033" height="189" alt="Screenshot 2025-10-21 at 9 04 24 PM" src="https://github.com/user-attachments/assets/57b31241-256e-4225-8218-0fff693085df" />
<img width="833" height="191" alt="Screenshot 2025-10-21 at 9 04 28 PM" src="https://github.com/user-attachments/assets/c0f50026-0842-4c17-803c-e8d24ff85a67" />
<img width="813" height="189" alt="Screenshot 2025-10-21 at 9 04 31 PM" src="https://github.com/user-attachments/assets/a76454a4-039e-4476-837e-4beca07caec0" />
<img width="756" height="186" alt="Screenshot 2025-10-21 at 9 04 35 PM" src="https://github.com/user-attachments/assets/c7240dd4-aa03-473a-9a31-77bcb68e1bb7" />
<img width="796" height="208" alt="Screenshot 2025-10-21 at 9 04 38 PM" src="https://github.com/user-attachments/assets/cd1910f2-cda4-40f6-ada0-0d2fb0961009" />
<img width="766" height="176" alt="Screenshot 2025-10-21 at 9 04 42 PM" src="https://github.com/user-attachments/assets/390a1e6d-8693-47b4-bbd7-307829132e82" />

# Speaker Cards

<img width="1666" height="2396" alt="gde_wtm_speakers" src="https://github.com/user-attachments/assets/b37dd796-1188-496a-8954-ac93f1a6a05b" />

# Partners Text

<img width="1630" height="212" alt="partners_text" src="https://github.com/user-attachments/assets/30ee43cd-6556-4d6b-ad92-0f0bb70b843a" />

# Speaker URLs

<img width="544" height="467" alt="Screenshot 2025-10-22 at 5 29 33 PM" src="https://github.com/user-attachments/assets/737e18be-d61a-4607-b94c-fe8f2131c97b" />
<img width="458" height="201" alt="Screenshot 2025-10-22 at 5 32 32 PM" src="https://github.com/user-attachments/assets/1e1fd382-473e-40c6-8206-6db624823111" />

## Additional Notes

<!-- Any additional information that reviewers should know -->

---

**Note**: This PR will be automatically checked for:

- Code quality (ESLint)
- Code formatting (Prettier)
- Build verification
- Accessibility compliance
- Security vulnerabilities

All checks must pass before this PR can be merged.

ShugKnight24 committed:
refactor: extract out ProfileCard

ShugKnight24 committed:
refactor: consolidate different TwitterHandle components

ShugKnight24 committed:
refactor: replace old CloseIcon

- Make it configurable

ShugKnight24 committed:
refactor: allow ProfileCard to handle speakers as well

- Utilize ProfileCard instead of GenericCard in SpeakerCard

ShugKnight24 committed:
chore: update speaker images with ImageMagik

- Make them fit with the new ProfileCards

ShugKnight24 committed:
chore: optimize speaker pictures with ImageOptim

ShugKnight24 committed:
refactor: add a placeholder when no sessions for a given track

- Add session descriptions

ShugKnight24 committed:
chore: add a bio

ShugKnight24 committed:
refactor: clarify & simplify logic

ShugKnight24 committed:
chore: remove extra space

ShugKnight24 commented on src/components/ui/ProfileCard.jsx:
```diff
@@ -0,0 +1,150 @@
+  const ribbonLabel = getRibbonLabel()
+
+  const isSpeaker = Boolean(track)
+  const teamBioColors = `bg-primary-500  hover:bg-primary-600`
```
resolved by [6e9a37a](https://github.com/GDG-Detroit/devfest-website/pull/81/commits/6e9a37a3e4251520e3e5ec0ad8308667608e7087)

ShugKnight24 committed:
fix: address copilot suggestions

ShugKnight24 commented on src/components/sessions/NoSessionsAvailable.jsx:
```diff
@@ -0,0 +1,32 @@
+
+NoSessionsAvailable.propTypes = {
+  currentSession: PropTypes.string.isRequired,
+}
```
resolved by [4eca416](https://github.com/GDG-Detroit/devfest-website/pull/81/commits/4eca41600db74b2bc32d8a3a3053ef6c1d00a49f)

ShugKnight24 commented on src/components/ui/ProfileCard.jsx:
```diff
@@ -0,0 +1,150 @@
+        {/* Ribbon Label */}
+        {ribbonLabel && (
+          <div
+            className={`ribbon-${devfest} absolute -right-10 top-6 z-10 w-40 rotate-45 ${ribbonColor} py-1 text-center text-xs font-semibold uppercase tracking-wide text-white shadow-md`}
```
resolved by [4eca416](https://github.com/GDG-Detroit/devfest-website/pull/81/commits/4eca41600db74b2bc32d8a3a3053ef6c1d00a49f)

ShugKnight24 committed:
chore: make bio consistent

ShugKnight24 commented on src/data/2025/team.js:
```diff
@@ -159,7 +159,11 @@ export const teamData = [
+    bio: `
+      Shugmi Shumunov is a software developer who believes code is the ultimate storytelling medium. With a primary focus on the JavaScript ecosystem within e-commerce, I've enjoyed working across the stack in various industries. I'm always seeking beautiful, elegant, and performant solutions while collaborating with talented, curious, and passionate engineers and tech enthusiasts.
+
+      When the terminal is closed and his hands are off the keyboard, you'll find him lifting weights, adventuring outdoors, exploring new places, or immersed in the worlds of manga or a good book
```
Resolved by [74b1138](https://github.com/GDG-Detroit/devfest-website/pull/81/commits/74b1138ac5d26b783d3109843bb584fb2fab3594)

shrinkray commented:
@ShugKnight24 I like the direction. That card template is not a good fit for the speakers, in my opinion. The card is very utilitarian, it's not designed for marketing and promotion. 

- the speaker image should be bigger
- Instead of a ribbon, there might be a different  type of badge or identifier
- I like the button
- I'm not saying you need to build this (see attached file), but the emphasis is on the image and it's stacked vertically. 

You can still wrap it in a box. (See attached) This code comes from TailwindCSS and is built in React. It has some simple data included. 

<img width="1270" height="1123" alt="Screenshot 2025-10-21 at 3 07 01 PM" src="https://github.com/user-attachments/assets/54b623dc-b59e-4273-9ce4-7dcebe658df1" />

[speakerMod.jsx.zip](https://github.com/user-attachments/files/23030175/speakerMod.jsx.zip)

shrinkray commented:
I like the coming soon stuff, that is spot on! One note, for the description, left align the text but offer it in a narrower box that is centered under the heading.

ShugKnight24 committed:
refactor: add latest speaker info

ShugKnight24 committed:
refactor: limit session description and align left

ShugKnight24 committed:
refactor: add startup track and update description

- alphabetize track descriptons

ShugKnight24 committed:
refactor: add additional speaker info

- Add new speaker assets
- Add placeholders for speakers without images

ShugKnight24 committed:
refactor: initial pass at improve profile cards for speakers

- Break card into atoms
- Make speaker card vertical
- Give speaker image a gradient and badge

ShugKnight24 commented:
# Updated Session text - contained to smaller size via padding & aligned left

<img width="1033" height="189" alt="Screenshot 2025-10-21 at 9 04 24 PM" src="https://github.com/user-attachments/assets/57b31241-256e-4225-8218-0fff693085df" />
<img width="833" height="191" alt="Screenshot 2025-10-21 at 9 04 28 PM" src="https://github.com/user-attachments/assets/c0f50026-0842-4c17-803c-e8d24ff85a67" />
<img width="813" height="189" alt="Screenshot 2025-10-21 at 9 04 31 PM" src="https://github.com/user-attachments/assets/a76454a4-039e-4476-837e-4beca07caec0" />
<img width="756" height="186" alt="Screenshot 2025-10-21 at 9 04 35 PM" src="https://github.com/user-attachments/assets/c7240dd4-aa03-473a-9a31-77bcb68e1bb7" />
<img width="796" height="208" alt="Screenshot 2025-10-21 at 9 04 38 PM" src="https://github.com/user-attachments/assets/cd1910f2-cda4-40f6-ada0-0d2fb0961009" />
<img width="766" height="176" alt="Screenshot 2025-10-21 at 9 04 42 PM" src="https://github.com/user-attachments/assets/390a1e6d-8693-47b4-bbd7-307829132e82" />

# Speaker Cards

<img width="1404" height="4076" alt="updated_speaker_cards" src="https://github.com/user-attachments/assets/95e0bed5-b8eb-45f1-ae0e-423e73524375" />

This will require some additional work. These are a little large and they are inconsistent when some of the speakers info is missing

shrinkray commented:
Big improvement. Thanks for your hard work here, and reconsidering the design. Here's my thoughts.

- increase font size on text areas & center 
- Details button can be moved up, and whole card can be linked instead of a button 
- since the card has bright colors, try to make the details button, same color as our CTAs white text. 
- the track color is good, I wonder about the color pallet, these color are a different hue and saturation?
- We could try a masonry layout with a card max height and let them be shorter, then the grid makes things more interesting 
- thanks for not using the ribbon, this is a more modern feel 
- the labels are like badges and might look better as pill shaped (ends half circles)
- Go for more rounded corners of the card

We need a description text below the 2025 heading, in a similar fashion to the developers area.

ShugKnight24 committed:
fix: replace `<li>`s with `<div>`s for accessibility

- Adjust card sizing across device sizes

ShugKnight24 committed:
refactor: update missing speaker images and data from LinkedIn

- Reoptimize speaker images w/ ImageOptim
- Joel Vasallo sent me a headshot

ShugKnight24 committed:
refactor: add a description to the speakers section

ShugKnight24 committed:
refactor: add a description to the partners section

ShugKnight24 committed:
refactor: break SpeakersContent into atoms

ShugKnight24 committed:
fix: replace `<ul>`s with `<div>`s for accessibility

ShugKnight24 committed:
refactor: update ProfileCard styling

- update colors
- update image aspect ration
- always display fields and clamp length to maintain consistency

ShugKnight24 commented:
~~@JustinBeaudry the Vercel deployment failed, can you let me know why and I can correct~~

All good

ShugKnight24 commented:
> Big improvement. Thanks for your hard work here, and reconsidering the design. Here's my thoughts.
> 
> * increase font size on text areas & center
> * Details button can be moved up, and whole card can be linked instead of a button
> * since the card has bright colors, try to make the details button, same color as our CTAs white text.
> * the track color is good, I wonder about the color pallet, these color are a different hue and saturation?
> * We could try a masonry layout with a card max height and let them be shorter, then the grid makes things more interesting
> * thanks for not using the ribbon, this is a more modern feel
> * the labels are like badges and might look better as pill shaped (ends half circles)
> * Go for more rounded corners of the card
> 
> We need a description text below the 2025 heading, in a similar fashion to the developers area.


@shrinkray thanks for the feedback, I made some adjustments accordingly

good call on the text, added to the partners section as well

- Not a fan of centering the text, think it looks poor in cases where it wraps - this is the issue that causes the inconsistencies with the cards. I'm constraining the number of lines, to prevent this but I don't really like having ellipses for the speaker's info and this still can cause minor layout shifts if some are 1 vs 2

- The whole card cannot be a button, there are nested interactive elements for the social links
- I updated the CTA
- I wasn't a fan of the colors, they were a placeholder, I updated some - I'm open to suggestions
- Masonry layout is fine, I can implement that if the group prefer, I opted for updating the amount of cards at different device sizes
- I made both the badge and the card more rounded though I'm not personally the biggest fan of that aesthetic

<img width="1630" height="212" alt="partners_text" src="https://github.com/user-attachments/assets/30ee43cd-6556-4d6b-ad92-0f0bb70b843a" />
<img width="1710" height="2366" alt="speakers" src="https://github.com/user-attachments/assets/efcb6ec5-de9a-42d9-8f71-af9a0eec355c" />

shrinkray commented:
@ShugKnight24  I see some things that would help on the card dimensions but it's not anything that would limit your progress on the PR. We should add a badge/indicator for anyone who is a GDG/WTM Expert. I'll introduce those ideas later. You've done a brilliant job pulling this together. 

People engage with rounded UI more than those with sharp corners. Honestly the CTAs might do better if they have softened corners as well (facepalm).

ShugKnight24 commented:
@shrinkray quick and dirty look at what masonry implementation with centered text and socials would look like

<img width="1873" height="2270" alt="masonry_center" src="https://github.com/user-attachments/assets/dd3f5640-4274-405b-855f-f6ba0f7e6d98" />

ShugKnight24 commented:
@shrinkray send me what specific images you were thinking. I'm already here, minor adjustments at this point

<img width="1592" height="838" alt="Screenshot 2025-10-22 at 12 20 48 PM" src="https://github.com/user-attachments/assets/dd107de5-86f9-4441-8c48-3615b6334deb" />

ShugKnight24 committed:
refactor: add latest photo of Adonai Vera

ShugKnight24 committed:
chore: update Jeseekia Vaughn's info

ShugKnight24 committed:
refactor: add WTM logo for panelists and icon for gdes

ShugKnight24 committed:
fix: prevent icons from overlapping with nav

ShugKnight24 commented:
Added icon for GDEs and Icons for WTM panelists

<img width="1666" height="2396" alt="gde_wtm_speakers" src="https://github.com/user-attachments/assets/28fcc63e-88df-4b8e-8e0b-01ff6782141f" />

shrinkray commented:
If the details button were moved up to the same row as the social media, then take back space on the bottom.

<img width="411" height="623" alt="move-details" src="https://github.com/user-attachments/assets/675f924d-7f17-4a89-b934-42ed0f984a0c" />

In dark mode, the gray text should be white, too—same with these cards in dark mode. 

<img width="475" height="243" alt="Screenshot 2025-10-22 at 3 29 39 PM" src="https://github.com/user-attachments/assets/0d5a8756-fa86-4f83-a94b-34deec1abeec" />

shrinkray commented on src/components/speakers/SpeakerCard.jsx:
This error returns in the console for props position, organization, and bio. 

Warning: Failed %s type: %s%s prop The prop `position` is marked as required in `SpeakerCard`, but its value is `null`. 
    at SpeakerCard (http://localhost:5173/src/components/speakers/SpeakerCard.jsx?t=1761160677032:16:24)
    at SpeakersContent (http://localhost:5173/src/components/speakers/SpeakersContent.jsx?t=1761160677033:18:28)
    at SpeakerProvider (http://localhost:5173/src/components/speakers/SpeakerContext.jsx:25:35)
    at SpeakersSection (http://localhost:5173/src/layouts/SpeakersSection.jsx:13:28)
    at main
    at PageLayout (http://localhost:5173/src/layouts/PageLayout.jsx:14:30)
    at Home
    at RenderedRoute (http://localhost:5173/node_modules/.vite/deps/react-router-dom.js?v=f438b190:3506:5)
    at Routes (http://localhost:5173/node_modules/.vite/deps/react-router-dom.js?v=f438b190:3888:5)
    at div
    at Router (http://localhost:5173/node_modules/.vite/deps/react-router-dom.js?v=f438b190:3835:15)
    at BrowserRouter (http://localhost:5173/node_modules/.vite/deps/react-router-dom.js?v=f438b190:4545:5)
    at App

ShugKnight24 replied:
Good catch. Ideally it should be, I suppose we can't guarantee we'll have all that info - I programmed defensively around it

ShugKnight24 replied:
resolved by [b28609a](https://github.com/GDG-Detroit/devfest-website/pull/81/commits/b28609afd6aab3110c6d632dfa5d57e3a1c36ce8)

ShugKnight24 committed:
refactor: adjust social and buttons to be inline for speaker

- dark mode adjustments

ShugKnight24 commented:
Agree, it looks better

<img width="1629" height="538" alt="Screenshot 2025-10-22 at 4 18 39 PM" src="https://github.com/user-attachments/assets/6d62a5c8-9ccb-44be-9901-192ed690b751" />

shrinkray reviewed (changes requested):
That error has me concerned. I've found other errors that are for things I built so I will be looking into those.

ShugKnight24 committed:
refactor: program defensively against potentially missing speaker data

ShugKnight24 committed:
fix: correct text color conflict between speaker and team cards

ShugKnight24 committed:
feat: add speaker url(s) to SpeakerDetails

shrinkray commented on src/components/ui/GenericCard.jsx:
In the speaker data (/src/data/2025/speakers.js), some speakers have position: null (e.g., speakers with id 14 and 17) or empty strings (e.g., speakers with id 1 and 4)
- GenericCard component has position marked as required in its PropTypes (line 91)?
- In speakers data, position can legitimately be `null` or an empty string
- The solution is to make the position prop optional in GenericCard's PropTypes definition, since the component already handles empty/null positions gracefully

shrinkray replied:
## Changes

- Update /Users/gregor/Code/REPOS/devfest-website/src/components/ui/GenericCard.jsx
- Change position: PropTypes.string.isRequired to position: PropTypes.string on line 91

This aligns with the actual data where speakers can have null positions (e.g., Alex Patterson, Stephen Wylie)
This simple change makes the PropTypes definition match the reality of the data and how the component actually works (it already displays empty positions gracefully).

ShugKnight24 replied:
Take a look where GenericCards are used - not with speakers
Those components are now deprecated and unused
Your previous error was around SpeakerCard

shrinkray replied:
Trying to show this as a diff, with this change I no longer see the error. 

<img width="965" height="283" alt="Screenshot 2025-10-22 at 5 45 47 PM" src="https://github.com/user-attachments/assets/037bf162-45aa-46ca-9afb-62abe9fa514a" />

shrinkray replied:
I'm not pushing that up but this was just a good way to show my test

ShugKnight24 replied:
I went ahead and fixed this anyway

Share the error you are seeing without those PropTypes

I will elaborate on my previous point - GenericCard is used in both FacilitatorsSection and the OrganizersSection
Neither of those use speakers. Furthermore, neither of those are used in the app any longer, I.E. GenericCards are no longer used on our app, so I'm curious how you are getting an error here

I'm fine making the adjustments

ShugKnight24 replied:
resolved by [6ad7e8a](https://github.com/GDG-Detroit/devfest-website/pull/81/commits/6ad7e8aa05752f571bbd28dd97353e557267c534)

ShugKnight24 commented:
We weren't displaying the speaker urls anywhere - I added them to the speaker details

<img width="544" height="467" alt="Screenshot 2025-10-22 at 5 29 33 PM" src="https://github.com/user-attachments/assets/e2e0f3c0-d369-4005-bd53-0018977e8606" />

<img width="458" height="201" alt="Screenshot 2025-10-22 at 5 32 32 PM" src="https://github.com/user-attachments/assets/1e1fd382-473e-40c6-8206-6db624823111" />

shrinkray reviewed (changes requested):
See the inline comments

ShugKnight24 committed:
refactor: defensively program around missing props in GenericCard

shrinkray commented:
Errors have been resolved.  This is looking good otherwise. Errors have disappeared. 

- At 1280px these might be wider

<img width="1048" height="510" alt="Screenshot 2025-10-22 at 6 14 41 PM" src="https://github.com/user-attachments/assets/d3ed74b8-1d0a-42e9-9c99-d86d812fd808" />

shrinkray reviewed (approved):
Errors are squashed, am only seeing layout issues; these can be addressed later.
